pypeit/spectrographs/gemini_gmos.py

```diff
@@ -641,37 +850,40 @@ def bpm(self, filename, det, shape=None, msbias=None):
         _bpm_img = np.expand_dims(bpm_img, 0) if nimg == 1 else bpm_img
 
         # Get the binning
-        # TODO: By definition this line means that filename cannot be None.
         # TODO: We're opening the file too many times...
-        hdu = io.fits_open(filename)
-        # TODO: Why aren't we usig get_meta_value for this?
-        binning = hdu[1].header['CCDSUM']
-        xbin = int(binning.split(' ')[0])
-        hdu.close()
+        hdrs = self.get_headarr(filename)
+        binning = self.get_meta_value(hdrs, 'binning')
+        mjd = self.get_meta_value(hdrs, 'mjd')
+        bin_spec, bin_spat= parse.parse_binning(binning) 
 
         # Add the detector-specific, hard-coded bad columns
         if 1 in _det:
             msgs.info("Using hard-coded BPM for det=1 on GMOSs")
             i = _det.index(1)
             # Apply the mask
-            badc = 616//xbin
+            badc = 616//bin_spec
             _bpm_img[i,badc,:] = 1
         if 2 in _det:
             msgs.info("Using hard-coded BPM for det=2 on GMOSs")
             i = _det.index(2)
             # Apply the mask
             # Up high
-            badr = (902*2)//xbin # Transposed
-            _bpm_img[i,badr:badr+(3*2)//xbin,:] = 1
+            badr = (902*2)//bin_spec # Transposed
+            _bpm_img[i,badr:badr+(3*2)//bin_spec,:] = 1
             # Down low
-            badr = (161*2)//xbin # Transposed
+            badr = (161*2)//bin_spec # Transposed
             _bpm_img[i,badr,:] = 1
+            # Bad amp as of January 28, 2022
+            # https://gemini.edu/sciops/instruments/gmos/GMOS-S_badamp5_ops_3.pdf
+            if mjd > 2022.07:
```

Comment:
This is confusing.  2022.07 is not an MJD.  According to the header keyword we're reading it's the epoch of the observation.  Can we rename the variable?

Reply:
done